### PR TITLE
refactor(test_lib): do not execute jasmine test as async if not required

### LINKED ISSUE #3893: [test_lib (JS)] jasmine.it() should execute asynchronously only when strictly required
We need to wrap jasmine `(x/i)it` functions so that we can use the test injector in angular tests.

The [current mechanism]( https://github.com/angular/angular/blob/be0739085976bbdbcf8f560475e87b07343f464d/modules/angular2/src/test_lib/test_lib.ts#L131) always pass to jasmine a callback with one parameter, ie `it('...', (done) => { ... });`.

When an `AsyncTestCompleter` is injected, calling the `done()` method on the instance forward to call to the `done` parameter provided by jasmine.

When no `AsyncTestCompleter` is injected, the angular specific wrapper will take care of calling `done()`.

As discussed with @IgorMinar, it would be better to not pass the `done` parameter unless striclty required (that is the test is async).

### PR DESCRIPTION
**Pull request by vicb**
fixes #3893 

@IgorMinar seems like this has no (positive) impact on test perf.

### CODE REVIEW

File: modules/angular2/src/test_lib/test_lib.ts [hunk 1]
```diff
@@ -54,19 +52,23 @@ var inIt = false;
...
 
-  beforeEach(fn: FunctionWithParamTokens) { this._fns.push(fn); }
+  beforeEach(fn: FunctionWithParamTokens | Function): void { this._fns.push(fn); }
 
-  run(injector) {
+  run(injector): void {
     if (this._parent) this._parent.run(injector);
-    this._fns.forEach((fn) => fn.execute(injector));
+    this._fns.forEach((fn) => {
+      return isFunction(fn) ? fn() : (<FunctionWithParamTokens>fn).execute(injector);
```
Comment (IgorMinar) [inline modules/angular2/src/test_lib/test_lib.ts]: typechecker is not happy about this line

**Comment by IgorMinar** [comment]
there is a typo in the commit message (refactot -> refactor)

the rest looks good.

Most of our tests are async, so I wouldn't expect to see a speedup in our test suite because of this change.

**Comment by vicb** [comment]
Fixed. Travis failure is unrelated to this PR.

**Comment by IgorMinar** [comment]
Can you rebase on master? that should make the travis failure go away.

**Comment by vicb** [comment]
Travis is happy now

**Comment by vicb** [comment]
@IgorMinar rebase (there were some conflicts due to recent renamings).

**Comment by vicb** [comment]
landed as 44a991e

**Pull request merged**